public/app/core/services/echo/backends/analytics/RudderstackBackend.ts

```diff
@@ -0,0 +1,74 @@
+import $ from 'jquery';
+import { EchoBackend, EchoEventType, isInteractionEvent, isPageviewEvent, PageviewEchoEvent } from '@grafana/runtime';
+import { User } from '../sentry/types';
+
+export interface RudderstackBackendOptions {
+  writeKey: string;
+  dataPlaneUrl: string;
+  user?: User;
+}
+
+export class RudderstackBackend implements EchoBackend<PageviewEchoEvent, RudderstackBackendOptions> {
+  supportedEvents = [EchoEventType.Pageview, EchoEventType.Interaction];
+
+  constructor(public options: RudderstackBackendOptions) {
+    const url = `https://cdn.rudderlabs.com/v1/rudder-analytics.min.js`;
+
+    $.ajax({
+      url,
+      dataType: 'script',
+      cache: true,
+    });
+
+    const rds = ((window as any).rudderanalytics = []);
+
+    var methods = [
+      'load',
+      'page',
+      'track',
+      'identify',
+      'alias',
+      'group',
+      'ready',
+      'reset',
+      'getAnonymousId',
+      'setAnonymousId',
+    ];
+
+    for (let i = 0; i < methods.length; i++) {
+      const method = methods[i];
+      (rds as Record<string, any>)[method] = (function (methodName) {
+        return function () {
+          // @ts-ignore
+          rds.push([methodName].concat(Array.prototype.slice.call(arguments)));
+        };
+      })(method);
+    }
+
+    (rds as any).load(options.writeKey, options.dataPlaneUrl);
+
+    if (options.user) {
+      (rds as any).identify(String(options.user.id), {
```

Comment:
I recommend changing this userId to be set to the email address of the user. 

Grafana+Rudderstack users will be able to (1) `alias` this userId to any other previously used identifier, plus (2) tools like customer.io will be able to automatically merge profiles that use multiple identifiers.

1. https://docs.rudderstack.com/stream-sources/rudderstack-sdk-integration-guides/rudderstack-javascript-sdk#3-5-alias
2. https://customer.io/docs/release-notes/enable-multiple-identifiers-in-your-current-workspace/

Reply:
@jmarbach thanks for input! Feel free to open a PR with the proposed change. Otherwise I'll do it next week.